README.md

```diff
@@ -121,84 +129,176 @@ export KEYDB_STORAGE_ACCESSKEY="your-secret-access-key"
 go run cmd/node/main.go
 ```
 
-### Scaling the Cluster
+## Scaling the Cluster
 
-To scale the KeyDB cluster, use the client's `CreateSnapshots`, `Scale` and `ScaleComplete` methods:
-Before scaling the cluster you should call `/createSnapshots` on the Scaler HTTP API to force all nodes to create
-snapshots of the hash ranges that need moving.
+To scale the KeyDB cluster, you can leverage the `Scaler` HTTP API.
 
-If the `CreateSnapshots` operation is skipped, new nodes added to the cluster during a scale operation, won't be
-able to get the data for the hash ranges that they are going to serve, leading to missing data.
+Internally the `Scaler` API uses the `internal/scaler` gRPC client to manage the cluster.
 
-```go
-// Create a client for scaler operations
-existingNodes := []string{
-    "localhost:50051",
-    "localhost:50052",
-    "localhost:50053",
-}
-scalerClient, err := client.NewClient(client.Config{
-    Addresses: existingNodes, // Connect to any existing node
-})
-if err != nil {
-    // Handle error
-}
-defer scalerClient.Close()
+### Scaling via HTTP API
 
-// Let's force a snapshot creation on the old nodes first
-err = scalerClient.CreateSnapshots(context.Background())
-if err != nil {
-    // Handle error
-}
+Here is an example of how to scale the cluster via the HTTP API:
+1. Use the `/hashRangeMovements` to preview a scaling operation
```

Comment:
We are only interested in figuring out how many hash ranges will be moved, correct? That is to assess how long the scaling operation will take?
```suggestion
1. Use the `/hashRangeMovements` to preview the number of hash movements of a scaling operation
```

Reply:
You can also "pre-upload" and "pre-download". I'll write it there to make it clear, thanks :+1: